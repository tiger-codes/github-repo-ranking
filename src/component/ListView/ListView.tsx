import React from "react";
import { ReactElement } from "react";
import ListItem from "../ListItem/ListItem";

interface Data {
    name: string;
    stargazers_count: number;
}

interface ListViewProps {
    data: Data[];
}

const ListView = ({ data }: ListViewProps ): ReactElement | null => {
    if (data.length === 0) return null;
    return data.map((item: Data) => <ListItem
                                        key={item.name}
                                        primaryText={item.name}
                                        secondaryText={item.stargazers_count}
                                    />) as unknown as ReactElement;
}

export default ListView;
