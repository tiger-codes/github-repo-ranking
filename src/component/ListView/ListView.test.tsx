import * as React from 'react';
import * as renderer from 'react-test-renderer';
import ListView from './ListView';

describe('<ListView />', () => {
  it('renders correctly', () => {
    const NoteComponent = renderer.create(<ListView data={[{ name: 'abc', stargazers_count:100 }]} />).toJSON();
    expect(NoteComponent).toMatchSnapshot();
  });
});
