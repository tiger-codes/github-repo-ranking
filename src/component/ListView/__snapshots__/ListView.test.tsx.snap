// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListView /> renders correctly 1`] = `
<div
  className="list-item-container"
>
  <div
    className="primary"
  >
    abc
  </div>
  <div
    className="secondary"
  >
    <span
      className="fa fa-star checked"
    />
    <span>
      100
    </span>
  </div>
</div>
`;
